File: JULY-2025/09-07-2025/p3439.erl
-import(lists,[sum/1,sublist/2,split/2]).
-import(erlang,[display/1]).

sub_lists([],[]) -> [];
sub_lists(List1,List2) -> [Head1|Tail1] = List1, [Head2|Tail2] = List2,
[Head1-Head2] ++ sub_lists(Tail1,Tail2).

mrt2(_,[],_) -> 0;
mrt2(Prev,Inc,Dec) -> [IncH|IncT] = Inc, [DecH|DecT] = Dec,
Curr = Prev + IncH - DecH, max(Curr,mrt2(Curr,IncT,DecT)).

max_running_total(List,K) when length(List) =< K -> sum(List);
max_running_total(List,K) -> {Head,Tail} = split(K,List), Sum=sum(Head), 
max(Sum,mrt2(Sum,Tail,List)).

-spec max_free_time(EventTime :: integer(), K :: integer(), StartTime :: [integer()], EndTime :: [integer()]) -> integer().
max_free_time(EventTime, K, StartTime, EndTime) ->
 Starts = StartTime ++ [EventTime], Ends = [0] ++ EndTime,
  Gaps = sub_lists(Starts,Ends), max_running_total(Gaps,K+1). 

File: JULY-2025/09-07-2025/p3439.c
int maxFreeTime(int eventTime, int k, int* startTime, int startTimeSize, int* endTime, int endTimeSize) {
    int gaps[startTimeSize+1];

    gaps[0] = startTime[0] - 0;
    for(int i=1; i<startTimeSize; i++){
        gaps[i] = startTime[i] - endTime[i-1];
    }
    gaps[startTimeSize] = eventTime - endTime[endTimeSize-1];

    int window_value=0;
    //first window
    for(int i=0; i<=k; i++){
        window_value += gaps[i];
    }
    int max = window_value;

    for(int i=(k+1); i<startTimeSize+1; i++){
        window_value -= gaps[i-(k+1)];
        window_value += gaps[i];
        if(window_value > max){
            max = window_value;
        }
    }

    return max;
}